C++/Examples/ADC/ADC.cpp

```diff
@@ -1,26 +1,55 @@
 #include <unistd.h>
 #include <cstdio>
-#include <Navio/Util.h>
-#include <Navio/ADC.h>
+#include <Common/Util.h>
+#include <Navio2/ADC.h>
+#include <Navio+/ADS1115.h>
+
 
 int main(int argc, char *argv[])
 {
-    ADC adc{};
-    adc.init();
-    float results[adc.get_channel_count()] = {0.0f};
-
     if (check_apm()) {
         return 1;
     }
 
-    while (true) {
-        for (int i = 0; i < ARRAY_SIZE(results); i++){
-            results[i] = adc.read(i);
-            printf("A%d: %.4fV ", i, results[i] / 1000);
+    if (get_navio_version() == NAVIO2) {
+
+        ADC adc{};
```

Comment:
Rename that to ADC_Navio2, add a new ADC interface and subtype. This is example code and should not deal with implementation details.